src/utils/tokens.ts

```diff
@@ -93,3 +101,77 @@ export const updateColonyTokens = async (
     }
   });
 };
+
+// Fetch all colonies that share the same native token
+export const fetchColoniesByNativeToken = async (
+  tokenAddress: string,
+): Promise<Array<Pick<FullColony, 'id' | 'status'>>> => {
```

Comment:
Alternatively this could be typed as:
```
Array<
    NonNullable<
      NonNullable<
        GetColonyByNativeTokenIdQuery['getColoniesByNativeTokenId']
      >['items'][number]
    >
  >
```
which is ugly but it would tie it to the actual query, right now this won't be updated if the queried fields change. Or you could create a new fragment for it